dll/win32/kernel32/client/proc.c

```diff
@@ -451,6 +466,66 @@ WINAPI
 BaseProcessStartup(
     _In_ PPROCESS_START_ROUTINE lpStartAddress)
 {
+    // for winsock inheritance
+    WSAPROTOCOL_INFOW ProtocolInfo;
+    WSADATA wsaData;
+    SOCKET sockDuplicated = INVALID_SOCKET;
+    PBYTE WinsockData = NULL;
+    ULONG sockindex = 0;
+    int sockloop;
+    int nStatus;
+    HMODULE winsock_module = NULL;
+    PPEB Peb = NtCurrentPeb();
+    typedef int (WINAPI *wsstcall)(DWORD, LPWSADATA);
+    typedef int (WINAPI *wsskcall)(int, int, int, LPWSAPROTOCOL_INFOW, GROUP, DWORD);
+    wsstcall DynWSAStartup;
+    wsskcall DynWSASocket;
+
+    RtlAcquirePebLock();
+    WinsockData = Peb->SparePtr2; // retreive PEB stored winsock inheritance data
+    sockindex = Peb->SpareUlong;
+    RtlReleasePebLock();
+
+    if (sockindex > 0) {
+        DPRINT("BaseProcessStartup sockindex retreived:%lx\n", sockindex);
+        DPRINT("BaseProcessStartup WinsockData pointer retreived:%p\n", WinsockData);
+	    winsock_module = LoadLibrary("WS2_32.DLL"); 
+  	    if (winsock_module)
+  	    {
+    		DynWSAStartup = (wsstcall)GetProcAddress(winsock_module, "WSAStartup");
+    		DynWSASocket = (wsskcall)GetProcAddress(winsock_module, "WSASocketW");
+	    } else {
+ 		    DPRINT1("BaseProcessStartup: winsock module faled to load it into child process.\n");
+	    }
+        if (winsock_module) 
+        {
+	        if ( DynWSASocket && DynWSAStartup ) 
+            {
+    	        if ((nStatus = DynWSAStartup(0x202,&wsaData)) == 0)
+    	        {
+		            for (sockloop = 0; sockloop < sockindex; sockloop = sockloop + 1) 
+                    {
+	            	    memcpy(&ProtocolInfo, &WinsockData[sockloop * ( sizeof(ProtocolInfo) )] , sizeof(ProtocolInfo));
+			            // NOTE: dwProviderReserved is used to pass the existing OS handle to WSASocket, since it is OBJ_INHERIT WSASocket should not create a new OS handle.
+	            	    sockDuplicated = DynWSASocket(FROM_PROTOCOL_INFO,
+        	                                            FROM_PROTOCOL_INFO,
+                	                                    FROM_PROTOCOL_INFO,
+	                                                    &ProtocolInfo,
+        	                                            0,
+                	                                    0);
+			            if (sockDuplicated == INVALID_SOCKET) 
+				            DPRINT1("BaseProcessStartup: WSASocket call failed to duplicate socket.\n");
+		            }
+    	         } else 
```

Comment:
```suggestion
    	         } else
```
EOL-whitespace. I do not point to each and everyone with suggestions, you can use the NPP feature to strip them all.